packages/api/src/index.ts

```diff
@@ -59,39 +64,48 @@ export const CardanoApi = {
   getOffChainMetadata,
   getOnChainMetadatas,
   getTokenSupply,
+  getProtocolParams,
   mockGetOnChainMetadatas,
   isNftMetadata,
   isFtMetadata,
   isMetadataFile,
+  cardanoApiMaker,
+  mockCardanoApi,
 } as const
 
-export namespace Cardano {
-  export namespace Api {
-    export type OffChainMetadataRequest = ApiOffChainMetadataRequest
-    export type OnChainMetadataRecord = ApiOnChainMetadataRecord
-    export type OffChainMetadataResponse = ApiOffChainMetadataResponse
+export namespace CardanoApi {
+  export type OffChainMetadataRequest = ApiOffChainMetadataRequest
+  export type OnChainMetadataRecord = ApiOnChainMetadataRecord
+  export type OffChainMetadataResponse = ApiOffChainMetadataResponse
 
-    export type OnChainMetadataRequest = ApiOnChainMetadataRequest
-    export type OffChainMetadataRecord = ApiOffChainMetadataRecord
-    export type OnChainMetadataResponse = ApiOnChainMetadataResponse
+  export type OnChainMetadataRequest = ApiOnChainMetadataRequest
+  export type OffChainMetadataRecord = ApiOffChainMetadataRecord
+  export type OnChainMetadataResponse = ApiOnChainMetadataResponse
 
-    export type TokenSupplyRequest = ApiTokeSupplyRequest
-    export type TokenSupplyRecord = ApiTokenSupplyRecord
-    export type TokenSupplyResponse = ApiTokenSupplyResponse
+  export type TokenSupplyRequest = ApiTokeSupplyRequest
+  export type TokenSupplyRecord = ApiTokenSupplyRecord
 
-    export type FutureToken = ApiFutureToken
-    export type FutureTokenRecords = ApiFutureTokenRecords
+  export type TokenIdentity = ApiTokenIdentity
+  export type TokenSupplyResponse = ApiTokenSupplyResponse
 
-    export type FtMetadata = ApiFtMetadata
-    export type FtMetadataRecord = ApiFtMetadataRecord
-    export interface FtRecords extends ApiFtRecords {}
-    export type TokenRegistryEntry = ApiTokenRegistryEntry
+  export type FutureToken = ApiFutureToken
+  export type FutureTokenRecords = ApiFutureTokenRecords
 
-    export type NftMetadata = ApiNftMetadata
-    export type NftMetadataRecord = ApiNftMetadataRecord
-    export interface NftRecords extends ApiNftRecords {}
+  export type FtMetadata = ApiFtMetadata
+  export type FtMetadataRecord = ApiFtMetadataRecord
+  export interface FtRecords extends ApiFtRecords {}
+  export type TokenRegistryEntry = ApiTokenRegistryEntry
 
-    export type MetadataFile = ApiMetadataFile
-    export type TokenId = ApiTokenId
+  export type NftMetadata = ApiNftMetadata
+  export type NftMetadataRecord = ApiNftMetadataRecord
+  export interface NftRecords extends ApiNftRecords {}
+
+  export type MetadataFile = ApiMetadataFile
+  export type TokenId = ApiTokenId
+
+  export type ProtocolParamsResult = ApiProtocolParamsResult
+
+  export interface api {
+    getProtocolParams: () => Promise<ProtocolParamsResult>
```

Comment:
I don't like this but I need it. Open to ideas.

Reply:
if its supposed to represent the module -> `CardanoApi.Module` -> export interface `Module` -> so the factory will be returning this object.